cfbs/commands.py

```diff
@@ -228,22 +229,52 @@ def init_command(index=None, non_interactive=False) -> int:
     """
     CFBSConfig.reload()
 
-    if prompt_user(
-        non_interactive,
-        "Do you wish to build on top of the default policy set, masterfiles? (Recommended)",
-        choices=YES_NO_CHOICES,
-        default="yes",
-    ) in ("yes", "y"):
-        to_add = "masterfiles"
-    else:
-        to_add = prompt_user(
+    branch = None
+    to_add = ""
+    if masterfiles is None:
+        if prompt_user(
             non_interactive,
-            "Specify policy set to use instead (empty to skip)",
-            default="",
-        )
+            "Do you wish to build on top of the default policy set, masterfiles? (Recommended)",
+            choices=YES_NO_CHOICES,
+            default="yes",
+        ) in ("yes", "y"):
+            to_add = "masterfiles"
+        else:
+            to_add = prompt_user(
+                non_interactive,
+                "Specify policy set to use instead (empty to skip)",
+                default="",
+            )
+    elif re.match(r"[0-9]+(\.[0-9]+){2}(\-[0-9]+)?", masterfiles):
+        log.debug("--masterfiles=%s appears to be a version number" % masterfiles)
+        to_add = "masterfiles@%s" % masterfiles
+    elif masterfiles != "no":
+        """This appears to be a branch. Thus we'll add masterfiles normally
+        and try to do the necessary modifications needed afterwards. I.e.
+        changing the 'repo' attribute to be 'url' and changing the commit to
+        be the current HEAD of the upstream branch."""
+
+        log.debug("--masterfiles=%s appears to be a branch" % masterfiles)
+        branch = masterfiles
+        to_add = "masterfiles"
 
     if to_add:
-        return add_command([to_add])
+        ret = add_command([to_add])
+        if ret != 0:
+            return ret
+
+    if branch is not None:
+        config = CFBSConfig.get_instance()
+        module = config.get_module_from_build("masterfiles")
+        remote = module["repo"]
+        commit = ls_remote(remote, branch)
+        if commit is None:
+            user_error("Failed to add masterfiles from branch %s" % branch)
+        log.debug("Current commit for masterfiles branch %s is %s" % (branch, commit))
+        module["url"] = remote
```

Comment:
Note / Todo: We should store the branch name here to enable `cfbs update`. But we can do this in a separate PR.

Reply:
[CFE-4089](https://tracker.mender.io/browse/CFE-4089)